File: AddAPearl.Domain/Product.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace AddAPearl.Domain
{
    public class Product
    {
        public int ProductId { get; set; }

        public string ProductName { get; set; }

        public string ProductCode { get; set; }

        public DateTime? ReleaseDate { get; set; }

        public decimal? Price { get; set; }

        public decimal PurchasePrice { get; set; }

        public DateTime? PurchaseDate { get; set; }

        public string Description { get; set; }

        public decimal? Rating { get; set; }
    }
}

File: AddAPearl.Web/src/app/shared/models/product.ts
import {
	IAddress,
	ICompany,
} from './index';

/* Defines the product entity */
export interface IProduct {
	productId: number;
	productName: string;
	productCode: string;
	releaseDate: string;
	price: number;
	purchasePrice: number;
	purchaseDate: string;
	description: string;
	rating: number;
}

export class Product implements IProduct {
	constructor(
		public productId: number,
		public productName: string,
		public productCode: string,
		public releaseDate: string,
		public price: number,
		public purchasePrice: number,
		public purchaseDate: string,
		public description: string,
		public rating: number,
	) {
		// Any other stuff to do during construction...
	}
}
